src/System.Private.CoreLib/shared/System/Numerics/Vector.cs

```diff
@@ -1319,67 +626,24 @@ public Vector(Span<T> values)
 
             if (Vector.IsHardwareAccelerated)
             {
-                if (typeof(T) == typeof(byte))
-                {
-                    for (int g = 0; g < Count; g++)
-                    {
-                        hash = HashCode.Combine(hash, ((byte)(object)this[g]).GetHashCode());
-                    }
-                    return hash;
-                }
-                else if (typeof(T) == typeof(sbyte))
-                {
-                    for (int g = 0; g < Count; g++)
-                    {
-                        hash = HashCode.Combine(hash, ((sbyte)(object)this[g]).GetHashCode());
-                    }
-                    return hash;
-                }
-                else if (typeof(T) == typeof(ushort))
-                {
-                    for (int g = 0; g < Count; g++)
-                    {
-                        hash = HashCode.Combine(hash, ((ushort)(object)this[g]).GetHashCode());
-                    }
-                    return hash;
-                }
-                else if (typeof(T) == typeof(short))
-                {
-                    for (int g = 0; g < Count; g++)
-                    {
-                        hash = HashCode.Combine(hash, ((short)(object)this[g]).GetHashCode());
-                    }
-                    return hash;
-                }
-                else if (typeof(T) == typeof(uint))
-                {
-                    for (int g = 0; g < Count; g++)
-                    {
-                        hash = HashCode.Combine(hash, ((uint)(object)this[g]).GetHashCode());
-                    }
-                    return hash;
-                }
-                else if (typeof(T) == typeof(int))
-                {
-                    for (int g = 0; g < Count; g++)
-                    {
-                        hash = HashCode.Combine(hash, ((int)(object)this[g]).GetHashCode());
-                    }
-                    return hash;
-                }
-                else if (typeof(T) == typeof(ulong))
+                if (typeof(T) == typeof(byte) ||
+                    typeof(T) == typeof(sbyte) ||
+                    typeof(T) == typeof(ushort) ||
+                    typeof(T) == typeof(short) ||
+                    typeof(T) == typeof(int) ||
+                    typeof(T) == typeof(uint))
```

Comment:
You could add _long_ and _ulong_ to this branch and treat them as two 32-bit integers for the sake of hash code computation; there's no need to keep them in their own branch. The only types that need to be treated specially are _float_ and _double_ since they have unique equality semantics.